### ISSUE TITLE
Empty server response, No servers available. Unable to fetch after new install.

### ISSUE THREAD

**Issue by mitchweaver** [aclist/dztui#141]
Hello! I am having an issue getting this to work on a new install, I've had this previously working before but its been a few months and now it is not. Hopefully I can provide enough info to help debug.

I don't think it's my steam API key as I have revoked and regenerated it a few times, all with the same effect. Plus it was originally doing this with my old key which was working previously. I have not deauthed my Steam Guard or changed username or any of the such that might temporarily make the API key stop working. Also waited 24 hours after my last revoke/regen and still no effect.

My guess is something on this new install / software versions is making this break.
Can confirm the servers result is empty after each request.

Happy to provide more info on request, cheers!

```
$ ls
dzg.cooldown  dzg.coords  dzg.servers
$ cat dzg.servers
$ file dzg.servers
dzg.servers: empty
$
```

DZGUI_SYSTEM.log: https://pastebin.com/raw/gD3ukcKp
DZGUI_DEBUG.log: https://pastebin.com/raw/jX0tsu5p

Distro: Gentoo
Kernel: 6.9.12
Python: 3.12.3
Bash: 5.2_p32
curl: 8.9.1-r1
jq: 1.7.1
zenity: 4.0.2
wmctrl: 1.07-r3
pygobject: 3.48.2-r1
dzgui: 6.3.2

**Comment by aclist** [comment]
Thanks for the detailed report. This seems to be happening to a handful of users recently, so I'm starting to think there may be more than meets the eye here and it is not just a random anomaly.

Do other Steam Web API queries work with that key?

Can you save the below as an executable script and run it to see if you get any output?

```
source $HOME/.config/dztui/dztuirc
limit=20000
url="https://api.steampowered.com/IGameServersService/GetServerList/v1/?filter=\appid\221100&limit=$limit&key=$steam_api"
curl -Ls "$url"
```

**Comment by mitchweaver** [comment]
@aclist Yep that script works just fine. API key confirmed good :)

**Comment by aclist** [comment]
Thanks for confirming. So something is happening later in the control flow. It might be a regression, although I can't reproduce it yet. Can you try this expanded version of the script above?

```
#!/usr/bin/env bash

parse_server_json(){
    local response="$1"
    <<< "$response" sed 's/\\u0001//g' | jq -r '
    .[]|"\(.name)␞" +
    "\(.map|if type == "string" then ascii_downcase else "null" end)␞" +
    "\(if .gametype == null then "null" else (.gametype|split(",")|if any(. == "no3rd") then "1PP" else "3PP" end) end)␞" +
    "\(if .gametype == null then "null" else (.gametype as $time|$time|test("[0-9]{2}:[0-9]{2}$") as $match|(if $match == true then ($time|scan("[0-9]{2}:[0-9]{2}$")) else "XXXX" end)) end)␞" +
    "\(.players)␞" +
    "\(.max_players)␞" +
    "\(if .gametype == null then "0" elif .gametype|split("lqs")[1] == null then "0" else .gametype|split("lqs")[1]|split(",")[0] end)␞" +
    "\(.addr|split(":")[0]):\(if .gameport == null then "XXXX" else .gameport end)␞" +
    "\(.addr|split(":")[1])"
    ' | sort -k1
}

source $HOME/.config/dztui/dztuirc
limit=20000
url="https://api.steampowered.com/IGameServersService/GetServerList/v1/?filter=\appid\221100&limit=$limit&key=$steam_api"
res=$(curl -Ls "$url" | jq -r '.response.servers')
parse_server_json "$res"
```

**Comment by aclist** [comment]
Also, do you know if your version of grep was compiled with Perl compatible regex support?

**Comment by mitchweaver** [comment]
Yes, have PCRE 
![image](https://github.com/user-attachments/assets/8d7e8e2f-cdd4-4b19-aafd-07f93f2bc76c)

but don't have oniguruma! That's the culprit!

![image](https://github.com/user-attachments/assets/0a50ae11-1b89-4d64-bf68-3f082fb32a0c)

Apparently that flag is default but I probably went "I don't know what that is, I don't use that!" and disabled it 🤣
Recompiled with it added and we are RESTORED! Server browser working again. 🙂

**Comment by mitchweaver** [comment]
Simple test could add dzgui to check for this:

```sh
if jq --build-configuration | grep oniguruma >/dev/null ; then
    echo yes
else
    echo no
fi
```

**Comment by aclist** [comment]
Thank you, I had a feeling this might be specific to gentoo (i.e. user compilation settings).

The fix is not quite so straightforward, though, because on most distributions the package will be linked against system packages/libraries and jq may not explicitly contain these compile flags. (For example, on Arch Linux, grepping jq's build configuration output will not pass the above test, despite being linked against the oniguruma package.)

I think this is more a case of gentoo being an outlier, so rather than test for the presence of oniguruma itself, we need to test for your ability to perform a basic parse with jq during the initial setup tests. We check for basic compatibility upfront, but it seems like it wouldn't hurt to make sure the parser is working.

When jq is compiled WITHOUT oniguruma, are you able to parse even basic JSON with it?

**Comment by mitchweaver** [comment]
Yes, haven't had an issue with any other programs or JSON. Specifically this is the error message:

`jq was compiled without ONIGURUMA regex libary. match/test/sub and related functions are not available.`

**Comment by aclist** [comment]
OK, this is great intel. During the filtration process, some of the queries indeed use the `test` function on the server JSON payload. 

In the event of failure (missing oniguruma dependency), does it return a non-zero exit code?

E.g.,

```
echo '{"test": "foo"}' | jq '.test | test("^foo$")'
echo $?
```

Obviously, gentoo users want a high degree of control over package compilation, so if something isn't working, they should be expected to have sane settings, but the cause of the error was not immediately clear, so I don't mind front-loading some extra validation here into DZGUI if it helps clarify the cause of the error so that users can set the right compile flags.

**Comment by aclist** [comment]
Alternatively, I can add this information to the documentation/knowledge base with a specific reminder for gentoo users.

**Comment by aclist** [comment]
Closing as resolved on Testing branch per 06359a4. Added pre-boot tests and extra warnings about self-compiled jq to documentation. Staged for imminent merge into Stable branch.
